src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpMetricSerializer.cs

```diff
@@ -0,0 +1,368 @@
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.InstrumentationScope_Version, meterVersion);
+        }
+
+        if (meterTags != null)
+        {
+            // TODO: DON'T HAVE ANY TESTS THAT EVALUATE THIS
+            foreach (var tag in meterTags)
+            {
+                writePosition = WriteTag(buffer, writePosition, tag, ProtobufOtlpMetricFieldNumberConstants.InstrumentationScope_Attributes);
+            }
+        }
+
+        ProtobufSerializer.WriteReservedLength(buffer, instrumentationScopeLengthPosition, writePosition - (instrumentationScopeLengthPosition + ReserveSizeForLength));
+
+        for (int i = 0; i < metrics.Count; i++)
+        {
+            writePosition = WriteMetric(buffer, writePosition, metrics[i]);
+        }
+
+        return writePosition;
+    }
+
+    private static int WriteMetric(byte[] buffer, int writePosition, in Metric metric)
+    {
+        writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ScopeMetrics_Metrics, ProtobufWireType.LEN);
+        int metricLengthPosition = writePosition;
+        writePosition += ReserveSizeForLength;
+
+        writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Name, metric.Name);
+
+        if (metric.Description != null)
+        {
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Description, metric.Description);
+        }
+
+        if (metric.Unit != null)
+        {
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Unit, metric.Unit);
+        }
+
+        switch (metric.MetricType)
+        {
+            case MetricType.LongSum:
+            case MetricType.LongSumNonMonotonic:
+            {
+                writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Data_Sum, ProtobufWireType.LEN);
+                int metricTypeLengthPosition = writePosition;
+                writePosition += ReserveSizeForLength;
+
+                writePosition = ProtobufSerializer.WriteBoolWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Sum_Is_Monotonic, metric.MetricType == MetricType.LongSum);
+                writePosition = ProtobufSerializer.WriteEnumWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Sum_Aggregation_Temporality, metric.Temporality == AggregationTemporality.Cumulative ? 2 : 1);
+
+                foreach (var metricPoint in metric.GetMetricPoints())
+                {
+                    var sum = metricPoint.GetSumLong();
+                    writePosition = WriteNumberDataPoint(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Sum_Data_Points, metricPoint, sum);
```

Comment:
When you pass around a `metricPoint` do so by ref:

```diff
writePosition = WriteNumberDataPoint(
    buffer,
    writePosition,
    ProtobufOtlpMetricFieldNumberConstants.Sum_Data_Points,
-   metricPoint,
+   in metricPoint,
    sum);
```

Seems to also have come from GenevaExporter 😄

Reply:
It isn't required at the call site. But not specifying it leaves it up to the compiler to decide if it needs to create a temporary copy or not. IMO, better to be explicit even if it might end up OK.

Some docs:
https://learn.microsoft.com/dotnet/csharp/language-reference/keywords/method-parameters#in-parameter-modifier